Sources/MapboxCoreNavigation/PassiveLocationManager.swift

```diff
@@ -248,34 +248,63 @@ open class PassiveLocationManager: NSObject {
+        self.sharedNavigator = sharedNavigator
+        self.directions = directions
+        self.systemLocationManager = systemLocationManager
+
+        super.init()
+
+        commonInit(directions: directions,
+                   systemLocationManager: systemLocationManager,
+                   eventsManagerType: eventsManagerType,
+                   userInfo: userInfo,
+                   datasetProfileIdentifier: datasetProfileIdentifier)
+    }
+
     /**
      Initializes the location manager with the given directions service.
-     
+
      - parameter directions: The directions service that allows the location manager to access road network data. If this argument is omitted, the shared value of `NavigationSettings.directions` will be used.
      - parameter systemLocationManager: The system location manager that provides raw locations for the receiver to match against the road network.
      - parameter eventsManagerType: An optional events manager type to use.
      - parameter userInfo: An optional metadata to be provided as initial value of `NavigationEventsManager.userInfo` property.
      - parameter datasetProfileIdentifier: custom profile setting, used for selecting tiles type for navigation. If set to `nil` - will not modify current profile setting.
-     
+
      - postcondition: Call `startUpdatingLocation()` afterwards to begin receiving location updates.
      */
     public required init(directions: Directions = NavigationSettings.shared.directions,
                          systemLocationManager: NavigationLocationManager? = nil,
                          eventsManagerType: NavigationEventsManager.Type? = nil,
                          userInfo: [String: String?]? = nil,
                          datasetProfileIdentifier: ProfileIdentifier? = nil) {
-        if let datasetProfileIdentifier = datasetProfileIdentifier {
-            Navigator.datasetProfileIdentifier = datasetProfileIdentifier
-        }
-
         self.sharedNavigator = Navigator.shared
-        
         self.directions = directions
-        
         self.systemLocationManager = systemLocationManager ?? NavigationLocationManager()
-        
+
         super.init()
+
+        commonInit(directions: directions,
+                   systemLocationManager: systemLocationManager,
+                   eventsManagerType: eventsManagerType,
+                   userInfo: userInfo,
+                   datasetProfileIdentifier: datasetProfileIdentifier)
+    }
+
+    private func commonInit(directions: Directions,
+                            systemLocationManager: NavigationLocationManager?,
+                            eventsManagerType: NavigationEventsManager.Type?,
+                            userInfo: [String: String?]?,
+                            datasetProfileIdentifier: ProfileIdentifier?) {
+        if let datasetProfileIdentifier = datasetProfileIdentifier {
+            Navigator.datasetProfileIdentifier = datasetProfileIdentifier
```

Comment:
This line still uses `Navigator` instead of `NavigatorProtocol`, which is undesired side effect. Also, this setup has to be done before `Navigator.shared` is created, otherwise it is useless.